static/src/javascripts/bootstraps/standalone.commercial.ts

```diff
@@ -154,19 +180,41 @@ const bootCommercial = async (): Promise<void> => {
 	try {
 		await loadFrontendBundle();
 		await loadDcrBundle();
-		await loadModules();
 
-		return catchErrorsWithContext(
-			[
+		// load just those modules required to display ads on the page
+		const baseModulesLoaded = loadBaseModules().then(() => {
+			catchErrorsWithContext(
 				[
-					'ga-user-timing-commercial-end',
-					function runTrackPerformance(): void {
-						EventTimer.get().trigger('commercialEnd');
-					},
+					[
+						'ga-user-timing-commercial-base-modules-loaded',
+						function runTrackPerformance(): void {
+							EventTimer.get().trigger(
+								'commercialBaseModulesLoaded',
+							);
+						},
+					],
 				],
-			],
-			tags,
-		);
+				tags,
+			);
+		});
```

Comment:
Could we keep the `await`s instead of `then()`s? Here I’m assuming we have loadModule take the modules as argument.

```ts
await loadModules(commercialBaseModules);

catchErrorsWithContext( /* … */)

await loadModules(commercialExtraModules);

catchErrorsWithContext( /* … */)
```

Reply:
Wouldn't this alter the order in which the operations are performed? My understanding is that currently base modules and remaining modules will be loaded simultaneously and a promise representing the completion of both is returned.
With the above suggestion, we'd load base modules first and only after they're loaded would we move on to the remaining ones. Wouldn't this increase the total time to load all commercial modules?

Reply:
We discussed this, and actually, there are three events we care about here:

1. base modules are loaded
2. extra modules are loaded
3. all commercial modules are loaded